cpp/src/io/avro/avro.cpp

```diff
@@ -21,6 +21,36 @@
 namespace cudf {
 namespace io {
 namespace avro {
+
+template <>
+uint64_t container::get_encoded()
+{
+  uint64_t val = 0;
+  for (uint64_t len = 0; len < 64; len += 7) {
+    auto const byte = get_raw<uint8_t>();
+    val |= (byte & 0x7f) << len;
+    if (byte < 0x80) break;
+  }
+  return val;
+}
+
+template <>
+int64_t container::get_encoded()
+{
+  auto const uval = get_encoded<uint64_t>();
+  return (int64_t)((uval >> 1u) ^ -(int64_t)(uval & 1));
+}
+
+template <>
+std::string container::get_encoded()
+{
+  auto len     = get_encoded<uint64_t>();
+  len          = ((len & 1) || (m_cur >= m_end)) ? 0 : std::min(len >> 1, (size_t)(m_end - m_cur));
```

Comment:
Avoid ITM and avoid C-style casts
```suggestion
  auto const len = [&] { 
    auto const len = get_encoded<uint64_t>();
    return (len & 1) || (m_cur >= m_end) ? 0 : std::min(len >> 1, static_cast<size_t>(m_end - m_cur));
  } ();
```